sycl/test-e2e/bindless_images/copies/copy_subregion_1D.cpp

```diff
@@ -544,42 +545,65 @@ bool run_copy_test(sycl::device &dev, sycl::queue &q, sycl::range<1> dims) {
                                             dataIn2, dev, q, out);
-
-  validated = validated && check_test(out, expected);
+  if (!check_test(out, expected)) {
+    std::cout << "copy_image_mem_handle_to_image_mem_handle test failed"
+              << std::endl;
+    validated = false;
+  }
 
   std::fill(out.begin(), out.end(), 0);
 
   copy_image_mem_handle_to_usm(dataInDesc, outDesc, dataIn1, dataIn2, dev, q,
                                out);
-
-  validated = validated && check_test(out, expected);
+  if (!check_test(out, expected)) {
+    std::cout << "copy_image_mem_handle_to_usm test failed" << std::endl;
+    validated = false;
+  }
 
   std::fill(out.begin(), out.end(), 0);
 
   copy_usm_to_image_mem_handle(dataInDesc, outDesc, dataIn1, dataIn2, dev, q,
                                out);
-
-  validated = validated && check_test(out, expected);
+  if (!check_test(out, expected)) {
+    std::cout << "copy_usm_to_image_mem_handle test failed" << std::endl;
+    validated = false;
+  }
 
   std::fill(out.begin(), out.end(), 0);
 
   copy_usm_to_usm(dataInDesc, outDesc, dataIn1, dataIn2, dev, q, out);
-
-  validated = validated && check_test(out, expected);
+  if (!check_test(out, expected)) {
+    std::cout << "copy_usm_to_usm test failed" << std::endl;
+    validated = false;
+  }
 
   // Perform out of bounds copy checks
-  validated =
-      validated && image_mem_handle_to_image_mem_handle_out_of_bounds_copy(
-                       dataInDesc, outDesc, dataIn1, dev, q);
+  if (!image_mem_handle_to_image_mem_handle_out_of_bounds_copy(
+          dataInDesc, outDesc, dataIn1, dev, q)) {
+    std::cout
+        << "image_mem_handle_to_image_mem_handle_out_of_bounds_copy test failed"
+        << std::endl;
+    validated = false;
+  }
 
-  validated = validated && image_mem_handle_to_usm_out_of_bounds_copy(
-                               dataInDesc, outDesc, dataIn1, dev, q);
+  if (!image_mem_handle_to_usm_out_of_bounds_copy(dataInDesc, outDesc, dataIn1,
+                                                  dev, q)) {
+    std::cout << "image_mem_handle_to_usm_out_of_bounds_copy test failed"
+              << std::endl;
+    validated = false;
+  }
 
-  validated = validated && usm_to_image_mem_handle_out_of_bounds_copy(
-                               dataInDesc, outDesc, dataIn1, dev, q);
+  if (!usm_to_image_mem_handle_out_of_bounds_copy(dataInDesc, outDesc, dataIn1,
+                                                  dev, q)) {
+    std::cout << "usm_to_image_mem_handle_out_of_bounds_copy test failed"
+              << std::endl;
+    validated = false;
+  }
 
-  validated = validated && usm_to_usm_out_of_bounds_copy(dataInDesc, outDesc,
-                                                         dataIn1, dev, q);
+  if (!usm_to_usm_out_of_bounds_copy(dataInDesc, outDesc, dataIn1, dev, q)) {
+    std::cout << "usm_to_usm_out_of_bounds_copy test failed" << std::endl;
+    validated = false;
+  }
 
   return validated;
```

Comment:
Alternatively, we could make `validated` an int, increment it on failure and return it. That could be the value returned by the main function too. Effectively also lets us count the number of failures. It is minor though.